akka-http-core/src/test/scala/akka/http/impl/engine/client/ConnectionPoolSpec.scala

```diff
@@ -228,6 +228,36 @@ class ConnectionPoolSpec extends AkkaSpec("""
       acceptIncomingConnection()
       val (Success(_), 42) = responseOut.expectNext()
     }
+
+    "never close hot connections when minConnections key is given and >0" in new TestSetup() {
+      // for lower bound of one connection
+      val minOneConnection = 1
+      val (_, _, _, hcpMinOneConnection) = cachedHostConnectionPool[Int](idleTimeout = 100.millis, minConnections = minOneConnection)
+      val gatewayOneConnection = hcpMinOneConnection.gateway
+
+      condHolds(1200.millis) { () ⇒
+        Await.result(gatewayOneConnection.poolStatus(), 300.millis).get shouldBe a[PoolInterfaceRunning]
+      }
```

Comment:
Ok, let's have it this way for now - until we have it introspectable.
1 second is likely a bit much